src/main/java/uk/gov/justice/probation/courtcasematcher/model/domain/CourtCase.java

```diff
@@ -1,98 +1,99 @@
 package uk.gov.justice.probation.courtcasematcher.model.domain;
 
 import com.fasterxml.jackson.annotation.JsonIgnore;
-import com.fasterxml.jackson.annotation.JsonInclude;
 import lombok.AccessLevel;
 import lombok.AllArgsConstructor;
 import lombok.Builder;
 import lombok.Data;
 import lombok.NoArgsConstructor;
 import lombok.Setter;
-import reactor.util.StringUtils;
+import lombok.With;
 
 import java.io.Serializable;
 import java.time.LocalDate;
 import java.time.LocalDateTime;
 import java.util.List;
 import java.util.Optional;
 
+import static org.springframework.util.StringUtils.hasText;
+
 @Data
 @Builder
+@With
 @AllArgsConstructor
 @NoArgsConstructor(access = AccessLevel.PRIVATE, force = true)
-@JsonInclude(JsonInclude.Include.NON_NULL)
 public class CourtCase implements Serializable {
 
     private final String caseId;
-
-    private final String defendantId;
-
     @Setter(AccessLevel.NONE)
     private final String caseNo;
+    private List<Defendant> defendants;
+    private List<HearingDay> hearingDays;
 
-    @Setter(AccessLevel.NONE)
-    private final String courtCode;
-
-    private final String courtRoom;
-
-    private final LocalDateTime sessionStartTime;
-
-    private final String probationStatus;
-
-    private final List<Offence> offences;
-
-    private final String crn;
-
-    private final String cro;
-
-    private final String pnc;
-
-    private final Name name;
-
-    private final String defendantName;
-
-    private final Address defendantAddress;
-
-    private final LocalDate defendantDob;
-
-    private final DefendantType defendantType;
-
-    private final String defendantSex;
-
-    private final String listNo;
-
-    private final String nationality1;
-
-    private final String nationality2;
-
-    private final Boolean breach;
+    @JsonIgnore
+    private final GroupedOffenderMatches groupedOffenderMatches;
 
-    private final LocalDate previouslyKnownTerminationDate;
+    private final DataSource source;
 
-    private final Boolean suspendedSentenceOrder;
+    public boolean shouldMatchToOffender() {
```

Comment:
Couldn't this be allowed to look at the entire list for now ? Then we won't have to change for CP. I see when starting the match you might only match the first for Libra. This may be a situation when I don't see th epicture in your head for the tickets though.

Reply:
It was a purely pragmatic decision, one less thing to think about. I know I'm not going to do the matching for all the defendants just yet so it made it easier to reason about if everything was only ever operating on the first defendant.  I began thinking about it then thought about whether this method will even belong on the CourtCase at all or whether it needs to live on the Defendant at which point I thought nah leave it for when that's your sole concern.